llvm/lib/MC/MCCASObjectV1.cpp

```diff
@@ -3024,3 +3024,170 @@ mccasformats::v1::loadDIETopLevel(DIETopLevelRef TopLevelRef) {
   return LoadedDIETopLevel{std::move(AbbrevData), *DistinctData,
                            *RootDIE};
 }
+
+struct DIEVisitor {
+  Error visitDIERef(DIEDataRef Ref);
+  Error visitDIERef(BinaryStreamReader &DataReader, unsigned AbbrevIdx,
+                    StringRef DIEData, ArrayRef<DIEDataRef> &DIEChildrenStack);
+  Error visitDIEAttrs(AbbrevEntryReader &AbbrevReader,
+                      BinaryStreamReader &Reader, StringRef DIEData);
+
+  ArrayRef<StringRef> AbbrevEntries;
+  BinaryStreamReader DistinctReader;
+
+  std::function<void(StringRef)> HeaderCallback;
+  std::function<void(dwarf::Tag, uint64_t)> StartTagCallback;
+  std::function<void(dwarf::Attribute, dwarf::Form, StringRef, bool)>
+      AttrCallback;
+  std::function<void(bool)> EndTagCallback;
+  std::function<void(StringRef)> NewBlockCallback;
+};
+
+Error DIEVisitor::visitDIEAttrs(AbbrevEntryReader &AbbrevReader,
+                                BinaryStreamReader &DataReader,
+                                StringRef DIEData) {
+  constexpr auto IsLittleEndian = true;
+  constexpr auto AddrSize = 8;
```

Comment:
Also the AddrSize, I am not sure if these will ever change, but maybe these shouldn't be constexprs